atlasdb-autobatch/src/test/java/com/palantir/atlasdb/autobatch/TracingEventHandlerTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.autobatch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.SettableFuture;
+import com.lmax.disruptor.EventHandler;
+import com.palantir.tracing.Observability;
+import com.palantir.tracing.Tracer;
+import com.palantir.tracing.api.Span;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TracingEventHandlerTest {
+
+    @Rule
+    public TraceCapturingRule traceRule = new TraceCapturingRule();
+
+    @Mock
+    private EventHandler<BatchElement<Integer, Long>> delegate;
+
+    @Test
+    public void nonFlushesDoNotHaveTraces() throws Exception {
+        TracingEventHandler<Integer, Long> tracingHandler =
+                new TracingEventHandler<>(delegate, "test", Observability.SAMPLE);
+
+        tracingHandler.onEvent(new TestBatchElement(), 45, false);
```

Comment:
Could you also add a pass-through test to assert that `tracingHandler.onEvent(x,y,z)` calls the correct delegate `handler`?

Reply:
I'll leave the addition of this test at your discretion.